sql/core/src/test/resources/sql-tests/inputs/join-lateral.sql

```diff
@@ -177,6 +177,25 @@ SELECT * FROM t3 JOIN LATERAL (SELECT EXPLODE_OUTER(c2));
 SELECT * FROM t3 JOIN LATERAL (SELECT EXPLODE(c2)) t(c3) ON c1 = c3;
 SELECT * FROM t3 LEFT JOIN LATERAL (SELECT EXPLODE(c2)) t(c3) ON c1 = c3;
 
+-- SPARK-41961: lateral join with table-valued functions
+SELECT * FROM LATERAL EXPLODE(ARRAY(1, 2));
```

Comment:
interesting, do other databases also support this syntax?

Reply:
Yes for example Postgres can run this query:
```
select * from lateral unnest(array[1, 2, 3]);
```